File: node_js/1_gem_pickup.js
/*
1 Gem Pickup example.

This is a simple example to complete the first training session.

Be sure you have started the 1 Gem Pickup training session and fill in the
bot_id and bot_token below. You may need to reset the training session if
you get any errors.

Be sure to install the node-fetch package as a module.
    npm install node-fetch

This example was written with node 20.10.0.
*/

import fetch from 'node-fetch';

// Fill in your bot ID and token.
const botID = "";
const botToken = "";

// You should not have to change either of these values.
const botHost = "https://engine.beeotee.com";
const headers = {Authorization: "Bearer " + botToken,
                 "Content-Type": "application/json"};

/*****************************************************************************
 * Get your bot info.
 ****************************************************************************/
const botsURL = botHost + "/bots/" + botID;
const botsResponse = await fetch(botsURL, { headers });
console.log(botsResponse.status + " " + botsURL);
const botsData = await botsResponse.json();
console.log(botsData);
const gridID = botsData.grid;
console.log("on grid: " + gridID);

// Wait 3 seconds between all calls.
await new Promise(r => setTimeout(r, 3000));

/*****************************************************************************
 * Get your grid info.
 * This includes the bot location.
 ****************************************************************************/
const gridsURL = botHost + "/grids/" + gridID;
const gridsResponse = await fetch(gridsURL, { headers });
console.log(gridsResponse.status + " " + gridsURL);
const gridsData = await gridsResponse.json();
console.log(gridsData);
// We need the gem id. There should only be 1 gem, so it is easy to find.
const gemID = gridsData.items[0].id;

// Wait 3 seconds between all calls.
await new Promise(r => setTimeout(r, 3000));

/*****************************************************************************
 * Move one space to the left.
 ****************************************************************************/
const moveLocation = {x: 0, y: 0};
const moveURL = botsURL + "/location";
const moveResponse = await fetch(moveURL, {
    method: "put",
    body: JSON.stringify(moveLocation),
    headers,
});
console.log(moveResponse.status + " " + moveURL);
const moveData = await moveResponse.json();
console.log(moveData);

// Wait 3 seconds between all calls.
await new Promise(r => setTimeout(r, 3000));

/*****************************************************************************
 * Pick up the gem.
 ****************************************************************************/
const pickupURL = botsURL + "/items/" + gemID;
const pickupResponse = await fetch(pickupURL, {
    method: "put",
    headers,
});
console.log(pickupResponse.status + " " + pickupURL);
const pickupData = await pickupResponse.json();
console.log(pickupData);

console.log("Congratulations!");
console.log("Refresh the web site to see if you've completed the training session.");

File: python/1_gem_pickup.py
"""
1 Gem Pickup example.

This is a simple example to complete the first training session.

Be sure you have started the 1 Gem Pickup training session and fill in the
bot_id and bot_token below. You may need to reset the training session if
you get any errors.

Be sure the requests package is installed.
    pip install requests

This example runs with Python 3.8 and above.
"""
from time import sleep
import requests

# Fill in your bot ID and token here.
bot_id = ""
bot_bearer_token = ""

# You should not have to change either of these values.
bot_host = "https://engine.beeotee.com"
headers = {"Authorization": f"Bearer {bot_bearer_token}"}

##############################################################################
# Get your bot info.
##############################################################################
url = f"{bot_host}/bots/{bot_id}"
response = requests.get(url, headers=headers)
content = response.json()
print(f"{response.status_code} {url}")
print(content)
assert response.status_code == 200
# Get the grid id.
grid_id = content["grid"]
print(f"on grid: {grid_id}")

# Wait 3 seconds between all calls.
sleep(3)

##############################################################################
# Get your grid info.
# This includes the bot location. If the bot is not at location 1,0,
# reset the training session.
##############################################################################
url = f"{bot_host}/grids/{grid_id}"
response = requests.get(url, headers=headers)
content = response.json()
print(f"{response.status_code} {url}")
print(content)
assert response.status_code == 200
# We need the gem id. There should only be 1 gem, so it is easy to find.
gem_id = content["items"][0]["id"]

# Wait 3 seconds between all calls.
sleep(3)

##############################################################################
# Move one space to the left.
##############################################################################
location = {"x": 0, "y": 0}
url = f"{bot_host}/bots/{bot_id}/location"
response = requests.put(url, json=location, headers=headers)
content = response.json()
print(f"{response.status_code} {url}")
print(content)
assert response.status_code == 200

# Wait 3 seconds between all calls.
sleep(3)

##############################################################################
# Pick up the gem.
##############################################################################
url = f"{bot_host}/bots/{bot_id}/items/{gem_id}"
response = requests.put(url, json=location, headers=headers)
content = response.json()
print(f"{response.status_code} {url}")
print(content)
assert response.status_code == 200

print("Congratulations!")
print("Refresh the web site to see if you've completed the training session.")
